kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -351,21 +352,38 @@ def retrieve(
                 url, params=query_params, headers=self.service.headers
             )
             response.raise_for_status()
+        except HTTPError as er:
+            # Try to use the id as it was given
+            if id.startswith(self.service.url_resources):
```

Comment:
So after few more tests, I think the nexus_path you had is needed for cross_bucket=True:

`nexus_path = f"{self.service.endpoint}/resources/"
          if not cross_bucket:
              nexus_path += f"{self.service.organisation}/{self.service.project}"
 `

` # Try to use the id as it was given, it might be a _self if it starts with self.service.url_resources
            if id.startswith(nexus_path):`

Reply:
I just tried too, you are correct, the nexus_path was still needed for the cross_bucket case.